gemrb/GUIScripts/pst/GUIREC.py

```diff
@@ -1191,17 +1192,22 @@ def GetThac0 (Class, Level):
 
 	return Thac0Table.GetValue (Class, str (Level))
 
-#apparently the original code doesn't follow profsmax/profs tables
-def HasGainedWeapProf (pc, currProf, currLevel, Class):
-	#only fighters gain weapon proficiencies
-	if Class!="FIGHTER":
-		return False
-	#hardcoded limit is 4
-	if currProf>3:
-		return False
-	if currProf > (currLevel - 1) / 3:
-		return False
-	return True
+# each gained level is checked for how many new prof points gained
+def GainedWeapProfs (pc, currProf, currLevel, AvatarName):
+
+	# Actually looking at the next level
+	nextLevel = currLevel + 1
+
+	# The table stops at level 20
+	if nextLevel < 21:
+		maxProf = CommonTables.CharProfs.GetValue(str(nextLevel), AvatarName)
+		return maxProf - currProf
+
+	# Nameless continues gaining points forever	at a rate of 1 every 3 levels
+	elif AvatarName == "NAMELESS" and (currProf-3) <= (nextLevel / 3):
```

Comment:
This should be a function instead: `AvatarName == "NAMELESS"`
It can also replace the `Specific == 2` checks.

Reply:
Good idea, should make the whole thing more readable